web-app/HeimScripts/heatmap/run.R

```diff
@@ -176,3 +189,85 @@ add.subset.label <- function(df,label){
   }
   return(df)
 }
+
+computeDendrogram <- function(zScoreMatrix, distanceMeassure, linkageMethod) {
+    as.dendrogram(hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod))
+}
```

Comment:
first hclust is run on a distance matrix not on zscore matrix, which is produced by running 'dist' on the z score matrix.
Distance matrix is then a matrix with the relative distance between each of the rows of data:
 dist(test)
         1        2
2 1.414214         
3 2.828427 1.414214
with test being:
     [,1] [,2]
[1,]    1    4
[2,]    2    5
[3,]    3    6


It partially contains 3x the same thing, because you run
hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod) for three different linkage methods. 
So the part 'dist(zScoreMatrix, method=distanceMeassure)' is calculated 3 times with the same input parameters: the same score data.frame and the same distance measure (euclidean or Manhattan)
Example: 
* colDendrogramEuclideanComplete <- computeDendrogram(t(measurements), 'euclidean', 'complete')
* colDendrogramEuclideanSingle <- computeDendrogram(t(measurements), 'euclidean', 'single')
* colDendrogramEuclideanAverage <- computeDendrogram(t(measurements), 'euclidean', 'average')
(Note: all are run using the transposed measurements matrix ('t(measurements)'), and with the same distance measure : euclidean distance)